object_detection/readme.md

```diff
@@ -0,0 +1,129 @@
+
+Colab offers free access to a computer that has reasonable GPU, it is a cloud service based on Jupyter Notebooks and internet connectivity is required for access.
+
+## Preprocessing
+
+The first step is to take pictures of the objects to train, after doing that, you have to reduce the resolution of the pictures (this reduces training time) and split the images into train/test folders.
+
+run the [transform_image_resolution.py](https://github.com/RoBorregos/Robocup-Home/tree/develop/object_detection/scripts/transform_image_resolution.py) script:
+```bash
+python transform_image_resolution.py -d ../images/ -s 800 600
+```
+- `-d` is the directory containing all the images.
+- `-s` is the resolution that will be applied to the images.
+
+run the [split_images](https://github.com/RoBorregos/Robocup-Home/tree/develop/object_detection/scripts/split_images.py) script.
+```bash
+python split_images.py -d ../images/complete_dataset -o ../images --train 80
+```
+- `-d` is the directory containing all the images.
+- `-o` is the directory where the train and test folder will be created.
+- `--train` is the percentage of images that will be used for training, 80% for train and %20 for test is the recommended. 
+
+Now you need to label the images, use the [labelimg](https://github.com/tzutalin/labelImg) open source tool to label all the pictures in both train/test directories (this is a tedious and long process).
+
+![labeling](https://github.com/RoBorregos/Robocup-Home/blob/object_detection/object_detection/images/readme_images/labeling.png)
+
+**As of this point you should have a `image` directory that contains your train and test images with respective xml file of each images.**
+
+Download the [generate_tfrecord.py](https://github.com/RoBorregos/Robocup-Home/tree/develop/object_detection/scripts/generate_tfrecord.py) script and change the `class_text_to_int` function by adding your own labels.
+```python
+def  class_text_to_int(row_label):
+	if row_label ==  'powerade':
+		return  1
+	elif row_label ==  'chocolate':
+		return  2
+	elif row_label ==  'dr_pepper':
+		return  3
+	elif row_label ==  'danup':
+		return  4
+	else:
+		None
+```
+
+Download the [labelmap](https://github.com/RoBorregos/Robocup-Home/blob/develop/object_detection/training/labelmap.pbtxt) file and change the `id` and `name` to your own labels. **NOTE:** Be consistent with the id you wrote on the `class_text_to_int` function.
+
+We are using the Faster-RCNN-Inception-V2 model. [Download the model here.](http://download.tensorflow.org/models/object_detection/faster_rcnn_inception_v2_coco_2018_01_28.tar.gz) Open the downloaded faster_rcnn_inception_v2_coco_2018_01_28.tar.gz file with a file archiver and extract the faster_rcnn_inception_v2_coco_2018_01_28 folder.
+
+Download the [faster_rcnn_inception_v2_pets](https://github.com/RoBorregos/Robocup-Home/blob/develop/object_detection/training/faster_rcnn_inception_v2_pets.config) file and change:
```

Comment:
I see this an almost exact version of the one in the .tar.gz but already adapted to this case.